### DISCUSSION TITLE
What is the best approach to using mikro-orm in a fastify project?

### DISCUSSION THREAD

**Discussion by manusanchev** [mikro-orm/mikro-orm#3520] [General]
Hi, I am trying to build a fastify project using mikro-orm, I followed this project as an example [https://github.com/mikro-orm/express-ts-example-app](https://github.com/mikro-orm/express-ts-example-app), but when I tried to use it in my route handler, I got a validation error.

ValidationError: Using global EntityManager instance methods for context specific actions is disallowed. If you need to work with the global instance's identity map, use `allowGlobalContext` configuration option or `fork()` instead.

**Comment by B4nan** [comment]
This is from a getting started guide I am currently working on:
```ts title='app.ts'
import { MikroORM, RequestContext } from '@mikro-orm/core';
import { fastify } from 'fastify';

export async function bootstrap(port = 3001) {
  const orm = await MikroORM.init();
  const app = fastify();

  // register request context hook
  app.addHook('onRequest', (request, reply, done) => {
    RequestContext.create(orm.em, done);
  });

  // shut down the connection when closing the app
  app.addHook('onClose', async () => {
    await orm.close();
  });

  // register routes here
  // ...

  const url = await app.listen({ port });

  return { app, url };
}
```

**Reply by manusanchev** [reply]
Thanks for the answer, that's what I was looking for.

**Comment by piotr-cz** [comment]
There is a [fastify-mikro-orm](https://github.com/serdnam/fastify-mikro-orm/) plugin, but it's code could be simplified

**Reply by bill-o-more** [reply]
it's also looks abandoned

**Reply by B4nan** [reply]
No need for any third party packages, its trivial to set things up:

https://github.com/mikro-orm/guide#fastify-setup

**Reply by bill-o-more** [reply]
Whoa, great guide! Doesn't a link to it belong in the end of the [Quick Start doc](https://mikro-orm.io/docs/quick-start)? :)

**Reply by B4nan** [reply]
Its not finished yet, it will be embedded to the docs once ready. Hard to find time for this during summer unfortunately.

**Reply by bill-o-more** [reply]
yeah, just got to the bottom and saw that it's a WIP. Still very impressive tho, docs of that grade are very rare :)

**Reply by bmingles** [reply]
I think you need to fork the em per request to avoid memory leaks right?

**Reply by bmingles** [reply]
Nvm, I see the RequestContext helper now
